src/checkbox/tests/unit/Checkbox.spec.tsx

```diff
@@ -0,0 +1,362 @@
+const { registerSuite } = intern.getInterface('object');
+const { assert } = intern.getPlugin('chai');
+
+import { tsx, create } from '@dojo/framework/core/vdom';
+import * as sinon from 'sinon';
+import harness from '@dojo/framework/testing/harness';
+
+import Label from '../../../label/index';
+import Checkbox from '../../index';
+import * as css from '../../../theme/checkbox.m.css';
+import { noop, stubEvent } from '../../../common/tests/support/test-helpers';
+import focus from '@dojo/framework/core/middleware/focus';
+
+const compareId = {
+	selector: 'input',
+	property: 'id',
+	comparator: (property: any) => typeof property === 'string'
+};
+const compareForId = {
+	selector: '@label',
+	property: 'forId',
+	comparator: (property: any) => typeof property === 'string'
+};
+
+const middlewareFactory = create();
+function createMockFocusMiddleware({
+	shouldFocus = false,
+	focused = false,
+	isFocused = false
+} = {}) {
+	return () =>
+		middlewareFactory(() => ({
+			shouldFocus: () => shouldFocus,
+			focused: () => focused,
+			isFocused: () => isFocused
+		}))();
+}
+
+const expected = function(label = false, checked = false) {
```

Comment:
This should really be using an assertion template

Reply:
Yeah, I just created a macro to convert what was there and didn't look too deeply at how the tests were structured. I went ahead and fixed that.